concepts/functions/about.md

```diff
@@ -0,0 +1,153 @@
+# About
+
+A function allows to group code into a reusable unit.
+There are multiple ways to define functions in JavaScript.
+Here we will look at function declarations and function expressions.
+Other possibilities like [arrow functions][concept-arrow-functions] will be covered in other concepts.
+
+## Function Declaration
+
+The standard way of defining a function in JavaScript is a function declaration, also called function definition or function statement.
+
+It consists of the `function` keyword, the name of the function and a comma-separated list of parameters in round brackets.
+This is followed by the function body (the code that should be executed) wrapped in curly brackets.
+
+```javascript
+function someName(param1, param2, param3) {
+  // ...
+}
+```
+
+## Using a Function
+
+In JavaScript a function is invoked (called) by stating the function name followed by round brackets that contain the arguments.
+
+```javascript
+someName(arg1, arg2, arg3);
+```
+
+Just stating the function name does **not** call the function in JavaScript.
+
+```javascript
+function sayHello() {
+  console.log('Hello, World!');
+}
+
+sayHello;
+// => [Function: sayHello]
+
+sayHello();
+// => 'Hello, World!'
+```
+
+## Parameters
+
+When working with parameters inside the function body, be aware of possible effects to the original value that was passed to the function.
+In JavaScript, the behavior depends on the data type of the argument.
+
+- All values that have primitive data types ([full list here][mdn-primitives]) are immutable in JavaScript, so if used as arguments, they are _passed by value_.
+  That means you are dealing with a copy of the original value in the function body and you can modify it without affecting the original value.
+- All other values (objects, arrays, functions) are _passed by reference_.
```

Comment:
Ah. Sorry, not. This is incorrect.

I previously [wrote an article](https://xpbytes.com/articles/javascript-is-not-call-by-reference/) about this _exact_ subject. Pedantically, in JavaScript *any value passed* is **always** _passed by value_, but the value that is passed in is _always a reference_ (even when it's a primitive). Primitive references can't be changed, which makes it seem like they are passed by their direct value (they're not). The "reference types" can be changed, which make it seem like they are passed by a reference. They are, but the reference itself is passed as... a value.

Confusing, I know. I think it's best to rewrite this and remove the terminology `by value` and `by reference`, optionally link to the article for those people that care. The reason I am suggesting this is because we _will_ have people from e.g. C do this track, and then the distinction matters.

Reply:
Good catch. I never looked into this in detail. I will read your article and then rephrase it.

Reply:
text updated as discussed on Slack